html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Root.pm

```diff
@@ -260,8 +260,18 @@ sub apply_new_node_info {
     my ( $status, $status_msg );
     ( $status, $status_msg ) = pf::node::node_register($self->current_mac, $self->username, %{$self->new_node_info()});
     if ($status) {
-        if($self->new_node_info->{category} && $self->new_node_info->{unregdate}) {
-            $self->app->flash->{notice} = [ "Role ".$self->new_node_info->{category}." has been assigned to your device with ". ( defined($self->new_node_info->{unregdate}) ? " unregistration date : ".$self->new_node_info->{unregdate} : "")."".( defined($self->new_node_info->{time_balance}) ? " time balance : ".$self->new_node_info->{time_balance} : "")."".( defined($self->new_node_info->{bandwidth_balance}) ? " bandwidth balance : ".$self->new_node_info->{bandwidth_balance} : "") ];
+        $self->app->flash->{notice} = "";
+        if($self->new_node_info->{category}) {
+            $self->app->flash->{notice} .= [ "Role %s has been assigned to your device : %s", $self->new_node_info->{category} ];
```

Comment:
You're concatenating an array

And you don't have enough variables for the amount of %s you have in the string